lib/CParser.mly

```diff
@@ -163,10 +165,10 @@ expr:
 | expr1 { $1 }
 
 expr0:
-| CONSTANT { Const(Constant.Concrete $1) }
-| CONSTVAR { Const(mk_sym $1) }
-| IDENTIFIER { LoadReg $1 }
-| STAR IDENTIFIER { LoadMem (LoadReg $2,AN []) }
+| CONSTANT { CBase.Const(Constant.Concrete $1) }
```

Comment:
We do need these as herd will fail to build on older versions of Ocaml (<=4.00)

error:
```
File "lib/CParser.mly", line 169, characters 13-18:
Warning 42: this use of Const relies on type-directed disambiguation,
it will not compile with OCaml 4.00 or earlier.
```
It's the same reason everything in lib/AArch64Parser now uses disambiguation